src/fparser/two/tests/fortran2003/test_hollerith_item.py

```diff
@@ -0,0 +1,139 @@
+from fparser.two.utils import NoMatchError, InternalError
+
+
+def test_hollerith(f2003_create, monkeypatch):
+    '''Check that a valid hollerith string is parsed correctly.'''
+    from fparser.two import utils
+    monkeypatch.setattr(utils, "EXTENSIONS", ["hollerith"])
+    for myinput in ["2Hab", "  2Hab", "1h ", "7h1234567"]:
+        ast = Hollerith_Item(myinput)
+        assert str(ast).upper() == myinput.lstrip().upper()
+
+
+def test_repr(f2003_create, monkeypatch):
+    '''Check that the repr output of a hollerith string gives the expected
+    result.
+
+    '''
+    from fparser.two import utils
+    monkeypatch.setattr(utils, "EXTENSIONS", ["hollerith"])
+    myinput = "2Hab"
+    ast = Hollerith_Item(myinput)
+    assert repr(ast) == "Hollerith_Item('ab')"
+
+
+def test_syntaxerror(f2003_create, monkeypatch):
+    '''test that an exception is raised if the supplied format is
+    invalid.
+
+    '''
+    from fparser.two import utils
+    monkeypatch.setattr(utils, "EXTENSIONS", ["hollerith"])
+    for myinput in [None, "", "  ", "0H", "1H", "2Hx" "2Hxxx", "H20", "xH",
+                    "2 Hxx"]:
+        with pytest.raises(NoMatchError):
+            _ = Hollerith_Item(myinput)
+
+
+def test_internal_error1(f2003_create, monkeypatch):
+    '''Check that an internal error is raised if the length of the Items
+    list is not 1 as the str() method assumes that it is.
+
+    '''
+    from fparser.two import utils
+    monkeypatch.setattr(utils, "EXTENSIONS", ["hollerith"])
+    myinput = "2Hab"
+    ast = Hollerith_Item(myinput)
+    monkeypatch.setattr(ast, "items", [None, None])
+    with pytest.raises(InternalError) as excinfo:
+        str(ast)
+    assert "should be of size 1 but found '2'" in str(excinfo)
+
+
+def test_internal_error2(f2003_create, monkeypatch):
+    '''Check that an internal error is raised if the string value (entry 0
+    of Items) is empty or None as the str() method assumes that it is
+    a string with content.
+
+    '''
+    from fparser.two import utils
+    monkeypatch.setattr(utils, "EXTENSIONS", ["hollerith"])
+    myinput = "2hab"
+    ast = Hollerith_Item(myinput)
+    for content in [None, ""]:
+        monkeypatch.setattr(ast, "items", [content])
+        with pytest.raises(InternalError) as excinfo:
+            str(ast)
+        assert ("entry 0 should be a valid Hollerith string but it is "
+                "empty") in str(excinfo)
+
+
+def test_invalid_hollerith(f2003_create, monkeypatch):
+    '''Test that the hollerith extension to the standard raises an
+    exception if it is not named as a valid extension.
+
+    '''
+    from fparser.two import utils
+    monkeypatch.setattr(utils, "EXTENSIONS", [])
+    myinput = "2Hab"
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = Hollerith_Item(myinput)
+        assert "Hollerith_Item: '{0}'".format(myinput) \
+            in str(excinfo.value)
+
+
+def test_valid_hollerith(f2003_create, monkeypatch):
```

Comment:
No, you didn't miss anything. You are right. It was a hangover from when I tested separately for EXTENSIONS having 'hollerith' specified or not. I then realised the other tests needed to set the value explicitly in case the default changes in the future